sdk/storage/azure-storage-blob/azure/storage/blob/_list_blobs_helper.py

```diff
@@ -7,14 +7,70 @@
 
 from azure.core.paging import PageIterator, ItemPaged
 from azure.core.exceptions import HttpResponseError
+
 from ._deserialize import get_blob_properties_from_generated_code, parse_tags
-from ._generated.models import BlobItemInternal, BlobPrefix as GenBlobPrefix, FilterBlobItem
+from ._generated.models import FilterBlobItem
 from ._models import BlobProperties, FilteredBlob
 from ._shared.models import DictMixin
+from ._shared.xml_deserialization import unpack_xml_content
 from ._shared.response_handlers import return_context_and_deserialized, process_storage_error
 
 
-class BlobPropertiesPaged(PageIterator):
+def deserialize_list_result(pipeline_response, *_):
+    payload = unpack_xml_content(pipeline_response.http_response)
+    location = pipeline_response.http_response.location_mode
+    return location, payload
+
+
+def load_xml_string(element, name):
+    node = element.find(name)
+    if node is None or not node.text:
+        return None
+    return node.text
+
+
+def load_xml_int(element, name):
+    node = element.find(name)
+    if node is None or not node.text:
+        return None
+    return int(node.text)
+
+
+def load_xml_bool(element, name):
+    node = load_xml_string(element, name)
+    if node and node.lower() == 'true':
+        return True
+    return False
+
+
+def load_single_node(element, name):
+    return element.find(name)
+
+
+def load_many_nodes(element, name, wrapper=None):
+    if wrapper:
+        element = load_single_node(element, wrapper)
+    return list(element.findall(name))
+
+
+def blob_properties_from_xml(element, select, deserializer):
+    if not select:
+        generated = deserializer.deserialize_data(element, 'BlobItemInternal')
+        return get_blob_properties_from_generated_code(generated)
+    blob = BlobProperties()
+    if 'name' in select:
+        blob.name = load_xml_string(element, 'Name')
+    if 'deleted' in select:
+        blob.deleted = load_xml_bool(element, 'Deleted')
+    if 'snapshot' in select:
+        blob.snapshot = load_xml_string(element, 'Snapshot')
+    if 'version' in select:
+        blob.version_id = load_xml_string(element, 'VersionId')
+        blob.is_current_version = load_xml_bool(element, 'IsCurrentVersion')
+    return blob
+
+
+class BlobPropertiesPaged(PageIterator):  # pylint: disable=too-many-instance-attributes
```

Comment:
It looks like I updated the existing BlobPropertiesPaged - which means the perf of this model would be improved, however if we wanted to leave the original `list_blobs` API _completely_ untouched, we could revert the changes here and have the new `list_blob_names` API use it's own custom Paged object.